operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -252,17 +284,41 @@ func (c *Controller) rateLimitProcessing() {
 	}
 }
 
+func (c *Controller) getQueue() workqueue.RateLimitingInterface {
+	var queue workqueue.RateLimitingInterface
+	c.cond.L.Lock()
+	defer c.cond.L.Unlock()
+	c.logger.Infof("Standard queue length: %d, fast queue length: %d", c.standardQueue.Len(), c.fastQueue.Len())
+	if c.fastQueue.Len() == 0 && c.standardQueue.Len() == 0 {
```

Comment:
we might need a for-loop here to block this as long as the queues are empty.

Reply:
Added Wait until condition to solve this